docs/src/developer/conventions.md

```diff
@@ -116,3 +117,17 @@ normalized as ``\frac{|\Omega|}{N} \sum_{r} |\psi(r)|^{2} = 1`` where
 and in reciprocal space its coefficients are ``\ell^{2}``-normalized,
 see the discussion in section [`PlaneWaveBasis` and plane-wave discretisations](@ref)
 where this is demonstrated.
+
+## Coding and formatting conventions
+The following conventions are not strictly enforced in DFTK. The rule of thumb is
+readability over consistency.
+
+- line lengths should be 92 characters.
+- avoid shortening variable names only for its own sake.
+- named tuples should be explicit, i.e., `(; var=val)` over `(var=val)`.
+- empty callbacks should use `identity`.
+- use `=` to loop over a range but `in` to loop over elements.
+- always format the `where` keyword with explicit braces: `where {T <: Any}`.
+- do not use implicit arguments in functions but explicit keyword.
+- prefix function name by `_` if it is an internal helper function, which is not of general
+  use and should thus be kept close to the vicinity of the calling function.
```

Comment:
Thanks for getting this started.

I think this deserves an extra page. We have many more, but they are just not documented yet.